### PR TITLE
Ward clus centers

### PR DESCRIPTION
**Pull request by brookehus**
- [x] Implement feature / fix bug
 - [x] Add tests
 - [x] Update changelog

Added `cluster_centers_` to `LandmarkAgglomerative` which computes the mean of all points in a cluster along every dimension

### CODE REVIEW

**Comment by brookehus** [comment]
I can get the double well test to work if I partition the double well into 4 bins along every dimension instead of 10 (see the last commit). I think 10 bins were getting zero populations/wonky transitions. Are @msmbuilder/developers okay with this change? Will assume so unless someone comments.

File: msmbuilder/cluster/agglomerative.py [hunk 1]
```diff
@@ -142,6 +142,8 @@ class _LandmarkAgglomerative(ClusterMixin, TransformerMixin):
     ----------
     landmark_labels_
     landmarks_
+    cluster_centers_ : mean of all points in each cluster along every dimension,
```
Comment (cxhernandez) [inline msmbuilder/cluster/agglomerative.py]: Does this mean you're using the centroid of all cluster centers?

File: msmbuilder/cluster/agglomerative.py [hunk 2]
```diff
@@ -260,6 +263,13 @@ def predict(self, X):
             else:
                 print("No data points were assigned to cluster {}".format(i))
 
+        if self.metric != 'rmsd':
+            cluster_centers_ = []
+            for i in range(self.n_clusters):
+                temp = list(np.mean(self.landmarks_[self.landmark_labels_==i],axis=0))
```
Comment (cxhernandez) [inline msmbuilder/cluster/agglomerative.py]: space before `axis`

File: msmbuilder/tests/test_agglomerative.py [hunk 3]
```diff
@@ -113,3 +113,13 @@ def test_alanine_dipeptide():
     labels = model.fit_predict(trajectories[0][0:100])
 
     assert len(np.unique(np.concatenate(labels))) <= n_clusters
+
+
+def test_cluster_centers():
+    x = [random.randn(20, 2) + 10, random.randn(20, 2)]
+    n_clusters = np.random.randint(2,7)
```
Comment (cxhernandez) [inline msmbuilder/tests/test_agglomerative.py]: space after comma

File: msmbuilder/cluster/agglomerative.py (hunk 1 above)
Comment (brookehus) [inline msmbuilder/cluster/agglomerative.py]: I don't understand your question- can you rephrase? For each cluster, I am taking the average along each dimension and reporting the 'center' as the point defined by those averages.

Comment (cxhernandez) [inline msmbuilder/cluster/agglomerative.py]: Ah okay, I misunderstood. Maybe just leave this as the default sklearn description for `cluster_centers_` (with a note about rmsd)? Something like:

>  Coordinates of cluster centers (unless RMSD is the metric)

Could you also follow the sklearn docstring template for attributes?

```
    Attributes
    ----------
    attribute_ : var_type, [dim1, dim2]
        Description here.
```

Comment (brookehus) [inline msmbuilder/cluster/agglomerative.py]: done 😃
